core/common/src/main/java/alluxio/retry/TimeBoundedRetry.java

```diff
@@ -52,22 +51,15 @@ public int getRetryCount() {
 
   @Override
   public boolean attemptRetry() {
-    if (mDone) {
-      return false;
-    }
     Instant now = mClock.instant();
-    // We should not do a retry if now == mEndTime. The final retry is timed to land at mEndTime,
-    // so if now == mEndTime, the operation may have taken less than 1ms.
-    if (!now.isBefore(mEndTime)) {
-      mDone = true;
+    if (now.isAfter(mEndTime) || now.equals(mEndTime)) {
```

Comment:
the `mDone` logic added a complexity just to handle an unlikely edge case where `attemptRetry` is called twice during the millisecond of `mEndTime`. Now we will return true twice in that case instead of just the first time.

Reply:
Won't we return `false` the first and second time now?

Reply:
oh right, the functionality is actually just the same, nvm my comment